### DISCUSSION TITLE
Feature: Alert User When A Contact Has Removed Signal

### DISCUSSION THREAD

**Discussion by muad-dumb** [AsamK/signal-cli#976] [Ideas]
My suggestion is some sort of alert-- such as a print-out after putting in a "receive" command --that informs you that someone on your Identities / Contacts list has removed Signal (i.e. can no longer receive encrypted messages).

I use a batch script to send one-way mass-texts, since Signal doesn't offer this and only offers Groups. An issue that I run into regularly is that someone on my batch file will have deleted / removed Signal, meaning that the next time I use Signal CLI with the batch script, I receive a "null [Unregistered User Exception]" message and only some of those messages are sent correctly, with the rest failing.

The resulting troubleshooting means a mix of; going through my Contacts on my Signal app, and opening a text message with every single contact until I find the one that no longer accepts encrypted messages; very occasionally I can use "listIdentities" and search around for a "null" user, which can be somewhat helpful, but the issue there is that there's no longer any identifying information that's useful in connecting the "null" Identity to the contact in my phone, or more importantly, with the defunct phone number.

Since Signal hasn't seen fit to implement this, I'm hoping that this community can pull something together!

**Comment by AsamK** [comment]
Not a notification, but there's the `getUserStatus` command that you could use to check which user isn't registered anymore.

**Reply by muad-dumb** [reply]
Oh, excellent! Is there a way to (or is that command capable of) printing Status for multiple identities / numbers at a time?

**Reply by AsamK** [reply]
Yeah, just call it with multiple numbers: `signal-cli getUserStatus +YYYY +ZZZZ`
